src/controller/artifact/annotation/v1alpha1/v1alpha1.go

```diff
@@ -0,0 +1,114 @@
+package v1alpha1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/goharbor/harbor/src/controller/artifact/annotation"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/artifact"
+	"github.com/goharbor/harbor/src/pkg/registry"
+
+	"github.com/docker/distribution/manifest/schema2"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const (
+	// V1alpha1 is the version of annotation parser
+	V1alpha1 = "v1alpha1"
+)
+
+func init() {
+	v1alpha1Parser := &v1alpha1Parser{
+		version: V1alpha1,
+		regCli:  registry.Cli,
+	}
+	if err := annotation.RegisterAnnotationParser(v1alpha1Parser, V1alpha1); err != nil {
+		log.Errorf("failed to register annotation parser for version %s: %v", V1alpha1, err)
+		return
+	}
+}
+
+type v1alpha1Parser struct {
+	version string
+	regCli  registry.Client
+}
+
+func (p *v1alpha1Parser) Parse(ctx context.Context, artifact *artifact.Artifact, manifest []byte) error {
```

Comment:
func is a bit long, I think we can use sub-functions(process extra attr, get layer process icon etc.) to complete the logic.